kedro/templates/project/hooks/utils.py

```diff
@@ -60,42 +63,127 @@ def setup_template_add_ons(selected_add_ons_list, requirements_file_path, pyproj
         pyproject_file_path: the path to the pyproject.toml file
             located on the the root of the template.
     """
+    # Checks if Linting is selected. If not, remove the associated lines from requirements and pyproject
     if "Linting" not in selected_add_ons_list:
-        pass
-    else:
-        with open(requirements_file_path, 'a') as file:
-            file.write(lint_requirements)
-        with open(pyproject_file_path, 'a') as file:
-            file.write(lint_pyproject_requirements)
-
+        with open(requirements_file_path, 'r') as file:
+            content = file.readlines()
+        with open(requirements_file_path, 'w') as file:
+            for line in content:
+                if line.strip() not in lint_requirements:
+                    file.write(line)
+
+        with open(pyproject_file_path, 'r') as file:
+            content = file.read()
+        with open(pyproject_file_path, 'w') as file:
+            content = content.replace(lint_pyproject_requirements, "")
+            file.write(content)
```

Comment:
I'm not quite sure I understand what's happening here. Is this checking the `requirements.txt` and `pyproject.toml` files from the starter and then rewriting only the content that's not linting? 

I think we should just make sure that the `pyproject.toml` and `requirements.txt` files in the starters contain the minimum so we can assume they don't contain any "add-ons" related stuff by default.

Reply:
Yes, as you said it rebuilding pyproject and requirements while removing the add-ons that are not selected. But if we are happy to strip pyproject.toml on starters then this won't be needed.

Reply:
I will revert it for now then.